features/admin.userstores.v1/components/user-stores-list.tsx

```diff
@@ -407,17 +416,15 @@ export const UserStoresList: FunctionComponent<UserStoresListPropsInterface> = (
 
         return [
             {
-                icon: (): SemanticICONS => hasRequiredScopes(featureConfig?.userStores,
-                    featureConfig?.userStores?.scopes?.update, allowedScopes) ?  "pencil alternate" : "eye",
-                onClick: (e: SyntheticEvent, userstore: UserStoreListItem): void => handleUserstoreEdit(userstore?.id),
-                popupText: (): string => hasRequiredScopes(featureConfig?.userStores,
-                    featureConfig?.userStores?.scopes?.update, allowedScopes) ? t("common:edit") : t("common:view"),
+                icon: (): SemanticICONS => hasUserStoreUpdatePermissions ?  "pencil alternate" : "eye",
+                onClick: (e: SyntheticEvent, userstore: UserStoreListItem): void =>
+                    handleUserstoreEdit(userstore?.id, userstore?.typeName),
+                popupText: (): string => hasUserStoreUpdatePermissions ? t("common:edit") : t("common:view"),
                 renderer: "semantic-icon"
             },
             {
                 hidden: (userstore: UserStoreListItem): boolean => {
-                    return !hasRequiredScopes(featureConfig?.userStores, featureConfig?.userStores?.scopes?.delete,
-                        allowedScopes) || userstore.id == CONSUMER_USERSTORE_ID || isPrivilegedUser;
+                    return !hasUserStoreUpdatePermissions || userstore.id == CONSUMER_USERSTORE_ID || isPrivilegedUser;
```

Comment:
Shouldn't this be delete permissions rather than update?

Reply:
Improved with 6b183b654a74dcf0ef25292a34d6ec9e2e620d04.